lib/kaffe/group_member/worker/worker.ex

```diff
@@ -0,0 +1,81 @@
+defmodule Kaffe.Worker do
+  @moduledoc """
+  A worker is assigned a single partition across topics. This is so we
+  effectively serialize the processing of any single key across topics.
+
+  Processing the message set is delegated to the configured message handler. It
+  is responsible for any error handling. The message handler must define a
+  `handle_messages` function (*note* the pluralization!) to accept a list of
+  messages.
+
+  The result of handle message is sent back to the subscriber.
+  """
+
+  alias Kaffe.Subscriber
+
+  require Logger
+  
+  @stat_messageset_process_time "subscriber.messageset.processed.time"
+
+  defmodule ListMessageHandler do
+    def handle_messages(messages, configured_handler) do
+      apply(configured_handler, :handle_messages, [messages])
+    end
+  end
+
+  defmodule IndividualMessageHandler do
+    def handle_messages(messages, configured_handler) do
+      Enum.each(messages, fn message ->
+        apply(configured_handler, :handle_message, [message])
+      end)
+    end
+  end
+
+  def start_link(message_handler, partition) do
+    GenServer.start_link(__MODULE__, [message_handler, partition], name: :"partition_worker_#{partition}")
+  end
+
+  def init([message_handler, partition]) do
+    Logger.info "event#starting=#{inspect self()} source=#{__MODULE__}"
+    {:ok, %{message_handler: determine_handler(message_handler),
+            configured_handler: message_handler,
+            partition: partition}}
+  end
+
+  def process_messages(pid, messages) do
+    GenServer.cast(pid, {:process_messages, self(), messages})
+  end
+
+  def handle_cast({:process_messages, subscriber_pid, messages},
+      %{message_handler: message_handler, configured_handler: configured_handler} = state) do
+
+    Metrix.measure @stat_messageset_process_time, fn ->
+      :ok = apply(message_handler, :handle_messages, [messages, configured_handler])
+      Subscriber.ack_messages(subscriber_pid)
+    end
+
+    {:noreply, state}
+  end
+
+  def terminate(reason, _state) do
+    Logger.info "event#terminate=#{inspect self()} reason=#{inspect reason}"
+  end
+
+  defp determine_handler(configured_handler) do
+    configured_handler.__info__(:functions)
```

Comment:
This feels too magical. I'd rather that end users declare the behavior they want directly rather than have it happen a result of their function names. What if they want to have both functions defined?